images/kubekins-e2e/kops-e2e-runner.sh

```diff
@@ -59,16 +59,6 @@ if [[ "${KOPS_DEPLOY_LATEST_KUBE:-}" =~ ^[yY]$ ]]; then
   e2e_args+=(--kops-kubernetes-version="${KOPS_KUBE_RELEASE_URL}/${KOPS_KUBE_LATEST}")
 fi
 
-EXTERNAL_IP=$(curl -SsL -H 'Metadata-Flavor: Google' 'http://metadata.google.internal/computeMetadata/v1/instance/network-interfaces/0/access-configs/0/external-ip' || true)
```

Comment:
Alternatively add a service account to this job

Reply:
It does have a service account preset defined [here](https://github.com/kubernetes/test-infra/blob/ac36d0bb3ca231bf9980186bd29a45a0a82af316/config/jobs/kubernetes/kops/kops-config.yaml#L62) and [this log](https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/directory/pull-kops-e2e-kubernetes-aws/1230550903434514432#1:build-log.txt%3A15) seems to indicate it is being used. Is that not sufficient?

Reply:
What does adding a `preset-service-account: "true"` label to the pod do here?

By service account I mean the pod needs a kubernetes service account. That will look something like this: https://github.com/kubernetes/test-infra/blob/ac36d0bb3ca231bf9980186bd29a45a0a82af316/config/jobs/kubernetes/test-infra/test-infra-trusted.yaml#L88

We declare the service accounts that exist for jobs in these two locations, depending on the cluster they run in:
https://github.com/kubernetes/test-infra/blob/ac36d0bb3ca231bf9980186bd29a45a0a82af316/prow/cluster/build_serviceaccounts.yaml#L7-L9

https://github.com/kubernetes/test-infra/blob/ac36d0bb3ca231bf9980186bd29a45a0a82af316/prow/cluster/trusted_serviceaccounts.yaml#L29-L31

The annotation controls which GCP service account metadata server (and thus the k8s service account) authenticates as

If the pod does not define a service account (default behavior) identity for its containers then GKE does not provide those container with a metadata server.

Reply:
The log is showing that the job has explicit credentials in a secret file passed to the job, which it uses for authentication (the secret contents of that file authenticates the user, the metadata server is not involved here).

Reply:
ah ok, so [this service account](https://github.com/kubernetes/test-infra/blob/ac36d0bb3ca231bf9980186bd29a45a0a82af316/config/prow/config.yaml#L503-L519) is separate from the serviceAccountName of the pod. That makes sense.

So we would be expected to create a new ServiceAccount resource in `build_serviceaccounts.yaml` correct? Which gcp-service-account would I want to use? Clearly this doesn't actually require GCP permissions, just access to the metadata server itself.

Reply:
Use `pr-kubekins@kubernetes-jenkins-pull.iam.gserviceaccount.com`, which is what the `secrets/service-account` file authenticates as. Maybe add a kops service account to that `build_serviceaccounts.yaml` file?

If you want to do this (which would be fantastic) then send me the PR. (Once it goes it you should be able to remove the `preset-service-account: true` label from your pods and not have to deal with the secret.json file any more too)